lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java

```diff
@@ -375,6 +379,55 @@ public Set<Location> listDirectories(Location location)
         });
     }
 
+    @Override
+    public Optional<Location> createTemporaryDirectory(Location targetLocation, String temporaryPrefix, String relativePrefix)
+            throws IOException
+    {
+        checkArgument(!relativePrefix.contains("/"), "relativePrefix must not contain slash");
```

Comment:
why isn't this also a requirement for azure?

Reply:
I don't validate it because Azure doesn't use it. In practice, this method is only called from one place with a constant value for relative prefix.